docs/excel/custom-functions-architecture.md

```diff
@@ -0,0 +1,50 @@
+---
+ms.date: 03/20/2019
+description: Learn about Excel custom functions' runtime. 
+title: Custom functions architecture (preview)
+localization_priority: Priority
+---
+# Custom functions architecture
+
+ Custom functions are with their own unique runtime that prioritizes execution of calculations. This article will cover the differences between the custom functions runtime and the browser-based JavaScript engine which powers most other parts of your add-in.
+
+## Custom functions runtime
+
+Because of their unique runtime, think of custom functions as another component of your total add-in. Your add-in will also likely include elements from the browser engine runtime, like a task pane.
+
+The following table highlights the differences between the custom functions runtime and the browser engine runtime:
+
+| Custom functions runtime 	| Browser engine runtime 	|
+|------------------------------------------------------------------	|--------------------------------------------------------------------------------------------------------------	|
+| Supports returning a value from a cell 	| Supports Office.js APIs and UI elements 	|
+| Does not have `localStorage` object, instead uses `AsyncStorage` 	| Has `localStorage` object, can optionally use `AsyncStorage` object 	|
+| Provides access to `OfficeRuntime` objects 	| Provides access to `OfficeRuntime` objects 	|
+| Allows you to make web requests 	| Allows you to access the Document Object Model (DOM) and supports libraries that use the DOM, such as jQuery 	|
+
+>[!NOTE]
```

Comment:
I would remove this note. Adam updated the template so you will always have a true task pane. However, this article is an excellent space to document the functions.html which loads custom-functions-runtime.js. This is the loader page to load your custom functions into the runtime and make the runtime JS available.